### DISCUSSION TITLE
Linting/formatting not working as expected with lint-staged

### DISCUSSION THREAD

**Discussion by viveleroi** [biomejs/biome#1611] [Q&A] [answered]
I've followed [the docs](https://biomejs.dev/recipes/git-hooks/#lint-staged) about setting biome up with lint-staged but am not sure it's working correctly. When lint-staged runs, I see the following output:

```
⚠ Detected incorrect braces with only single value: `**.{js|ts|cjs|mjs|d.mts|jsx|tsx|json|jsonc}`. Reformatted as: `**.js|ts|cjs|mjs|d.mts|jsx|tsx|json|jsonc`

→ No staged files match any configured task.
```

It seems like:

1. Biome is reformatting the example I copied directly from the biome docs, except it's not actually writing the file
2. The pattern isn't matching any files even though I had `tsx` files staged
3. Somehow biome is running yet lint-staged didn't match tasks? huh?

I have the following in my package.json:

```
"lint-staged": {
  "**.{js|ts|cjs|mjs|d.mts|jsx|tsx|json|jsonc}": ["biome check --apply --no-errors-on-unmatched"]
},
```

**Comment by ematipico** [comment]
It's possible we made a mistake. Can you try with just one `*`? E.g. `*.{js,jsx}`

**Reply by viveleroi** [reply]
Doesn't appear to affect the result:

```
⚠ Detected incorrect braces with only single value: `*.{js|ts|cjs|mjs|d.mts|jsx|tsx|json|jsonc}`. Reformatted as: `*.js|ts|cjs|mjs|d.mts|jsx|tsx|json|jsonc`

→ No staged files match any configured task.
```

**Reply by viveleroi** [reply, marked as answer]
It needs to use commas, which showed in your example but I originally missed - but the docs show pipes so they are wrong.

`"*.{js,ts,cjs,mjs,d.mts,jsx,tsx,json,jsonc}"`

**Reply by Conaclos** [reply]
Fixed. Thanks for your help!
